New Getting Started content for API Testing (#1388)
saucelabs/sauce-docs: docs/api-testing.md

---
id: api-testing
title: API Testing
sidebar_label: Getting Started
description: Running API tests with Sauce Labs
hide_table_of_contents: true
---

> At its core, Sauce Labs API Testing allows you to use either and HTTP client or a proprietary test composer to generate and edit intelligent contract or functional tests. The tests can be automated with any CI/CD tool or using the platform’s scheduler. Functional tests can then be used as load tests or scheduled as uptime monitors.

Reviewer: ```suggestion
At its core, Sauce Labs API Testing allows you to use an HTTP client or a proprietary test composer to generate and edit intelligent contract or functional tests. The tests can be automated with any CI/CD tool or using the platform’s scheduler. Functional tests can then be used as load tests or scheduled as uptime monitors.
```

<img src="/img/api-testing/API Testing Getting Started.png" width="600"/>
<div className="box-wrapper" markdown="1">

<div className="box box1 card">
<div className="container">
<h2>Contract Testing</h2>
<p>
</p>
<p>

> Use the API Testing platform to generate contract tests from an OpenAPI specification file. When used with Piestry, our mocking tool, you can test both the consumer and the provider.

Reviewer 2: ```suggestion
Use the API Testing platform to generate contract tests from an OpenAPI specification file. When used with Piestry, our mocking tool, you can test both the consumer and the producer side.
```

</p>
<ul>
<li><a href="/api-testing/contract-testing">API Contract Testing</a></li>
<li><a href="/api-testing/mocking">API Mocking with Piestry</a></li>
</ul>
</div>
</div>

<div className="box box2 card">
<div className="container">

<h2>Functional Testing</h2>
<p></p>
<p>
Create functional tests from multiple locations:
<ul>
<li>
<a href="/api-testing/quickstart">HTTP client</a></li>
<li>
<a href="/api-testing/import-postman-collection">Postman collection (in the HTTP client)</a></li>
<li>
<a href="/api-testing/build-from-spec">OpenAPI spec file or HAR file</a>
</li>
<li>

> <a href="/api-testing/import-postman-collection/#importing-a-har-file-from-an-rdc-job">HAR files captured during real device testing</a>

Reviewer 2: link above is no longer valid, now we have both RDC and VDC. Replace with:
<a href="/api-testing/import-postman-collection/#importing-a-har-file-rdc-and-vdc">Import from HAR file (including HAR files from RDC and VDC)</a>

</li>
<li>
<a href="/api-testing/composer">Test composer</a>
</li>
</ul>
</p>
</div>
</div>

<div className="box box3 card">
<div className="container">

<h2>Monitoring</h2>
<p></p>
<p>
Schedule existing functional tests against any environment for staging and production monitoring, and to validate the functional uptime of business critical APIs. See <a href="/api-testing/schedule-test">Scheduling API Tests</a> for more information.
</p>
</div>
</div>

<div className="box box4 card">
<div className="container">

<h2>Testing Private APIs</h2>
<p></p>
<p>

> Use Sauce Connect Proxy to test APIs behind a firewall. This launches a secured tunnel between the Sauce Labs API Testing platform and your APIs. See <a href="/secure-connections/sauce-connect">Sauce Connect Proxy</a> for more information.

Reviewer: ```suggestion
Use Sauce Connect Proxy to test APIs behind a firewall. This launches a secure tunnel between the Sauce Labs API Testing platform and your APIs. See <a href="/secure-connections/sauce-connect">Sauce Connect Proxy</a> for more information.
```

> Use Sauce Connect Proxy to test APIs behind a firewall. This launches a secured tunnel between the Sauce Labs API Testing platform and your APIs. See <a href="/secure-connections/sauce-connect">Sauce Connect Proxy</a> for more information.

Reviewer 2: ```suggestion
Use Sauce Connect Proxy to test APIs behind a firewall. This launches a secure tunnel between the Sauce Labs API Testing platform and your APIs. See <a href="/api-testing/sauce-connect/">Sauce Connect Proxy</a> for more information.
```
